File: javascript/167-two-sum-ii.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(numbers, target) {
    let lowerBound = 0, upperBound = numbers.length-1
    while (lowerBound < upperBound) {
        let sum = numbers[lowerBound] + numbers[upperBound]
        if (sum === target) return [lowerBound+1, upperBound+1]
        sum < target ? lowerBound++ : upperBound--
    }
};

File: golang/167-two-sum-ii.go
package golang

func twoSum(numbers []int, target int) []int {
    lowerBound, upperBound := 0, len(numbers)-1
    result := make([]int, 2)
	for lowerBound < upperBound {
		sum := numbers[lowerBound] + numbers[upperBound]
		if sum == target {
            result[0] = lowerBound+1
            result[1] = upperBound+1
            break
		} else if sum < target {
			lowerBound++
		} else {
			upperBound--
		}
	}
	return result
}